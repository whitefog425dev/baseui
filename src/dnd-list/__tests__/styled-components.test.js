/*
Copyright (c) 2018 Uber Technologies, Inc.

This source code is licensed under the MIT license found in the
LICENSE file in the root directory of this source tree.
*/
// @flow
import React from 'react';
import {shallow} from 'enzyme';
import {
  StyledRoot,
  StyledList,
  StyledItem,
  StyledDragHandle,
  StyledCloseHandle,
  StyledLabel,
} from '../index.js';

describe('Component styled components', () => {
  test('StyledRoot - basic render', () => {
    const component = shallow(<StyledRoot />);
    expect(component.instance().getStyles()).toMatchSnapshot(
      'StyledRoot has correct styles',
    );
  });
  test('StyledList - basic render', () => {
    const component = shallow(<StyledList />);
    expect(component.instance().getStyles()).toMatchSnapshot(
      'StyledList has correct styles',
    );
  });
  test('StyledList - dragging is active', () => {
    const component = shallow(<StyledList $isDragged />);
    expect(component.instance().getStyles()).toMatchSnapshot(
      'StyledList has correct styles when dragging is active',
    );
  });
  test('StyledItem - basic render', () => {
    const component = shallow(<StyledItem />);
    expect(component.instance().getStyles()).toMatchSnapshot(
      'StyledItem has correct styles',
    );
  });
  test('StyledItem - dragged', () => {
    const component = shallow(<StyledItem $isDragged />);
    expect(component.instance().getStyles()).toMatchSnapshot(
      'StyledItem has correct styles when being dragged',
    );
  });
  test('StyledItem - selected', () => {
    const component = shallow(<StyledItem $isSelected />);
    expect(component.instance().getStyles()).toMatchSnapshot(
      'StyledItem has correct styles when being selected',
    );
  });
  test('StyledDragHandle - basic render', () => {
    const component = shallow(<StyledDragHandle />);
    expect(component.instance().getStyles()).toMatchSnapshot(
      'StyledDragHandle has correct styles',
    );
  });
  test('StyledCloseHandle - basic render', () => {
    const component = shallow(<StyledCloseHandle />);
    expect(component.instance().getStyles()).toMatchSnapshot(
      'StyledCloseHandle has correct styles',
    );
  });
  test('StyledLabel - basic render', () => {
    const component = shallow(<StyledLabel />);
    expect(component.instance().getStyles()).toMatchSnapshot(
      'StyledLabel has correct styles',
    );
  });
});
