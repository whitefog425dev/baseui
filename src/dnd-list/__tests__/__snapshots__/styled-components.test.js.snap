// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component styled components StyledCloseHandle - basic render: StyledCloseHandle has correct styles 1`] = `
Object {
  "alignItems": "center",
  "backgroundColor": "transparent",
  "border": "none",
  "cursor": "pointer",
  "display": "flex",
  "marginLeft": "$theme.sizing.scale600",
  "overflow": "visible",
  "paddingBottom": 0,
  "paddingLeft": 0,
  "paddingRight": 0,
  "paddingTop": 0,
  "width": "$theme.sizing.scale800",
}
`;

exports[`Component styled components StyledDragHandle - basic render: StyledDragHandle has correct styles 1`] = `
Object {
  "alignItems": "center",
  "display": "flex",
  "marginRight": "$theme.sizing.scale600",
  "width": "$theme.sizing.scale800",
}
`;

exports[`Component styled components StyledItem - basic render: StyledItem has correct styles 1`] = `
Object {
  ":hover": Object {
    "borderBottomColor": "$theme.colors.primary100",
    "borderBottomStyle": "solid",
    "borderBottomWidth": "2px",
    "borderLeftColor": "$theme.colors.primary100",
    "borderLeftStyle": "solid",
    "borderLeftWidth": "2px",
    "borderRightColor": "$theme.colors.primary100",
    "borderRightStyle": "solid",
    "borderRightWidth": "2px",
    "borderTopColor": "$theme.colors.primary100",
    "borderTopStyle": "solid",
    "borderTopWidth": "2px",
    "color": "$theme.colors.primary",
  },
  "alignItems": "center",
  "backgroundColor": "$theme.colors.mono100",
  "borderBottomColor": "transparent",
  "borderBottomStyle": "solid",
  "borderBottomWidth": "2px",
  "borderLeftColor": "transparent",
  "borderLeftStyle": "solid",
  "borderLeftWidth": "2px",
  "borderRightColor": "transparent",
  "borderRightStyle": "solid",
  "borderRightWidth": "2px",
  "borderTopColor": "transparent",
  "borderTopStyle": "solid",
  "borderTopWidth": "2px",
  "boxShadow": null,
  "cursor": "grab",
  "display": "flex",
  "justifyContent": "space-between",
  "listStyle": "none",
  "paddingBottom": "$theme.sizing.scale600",
  "paddingLeft": "$theme.sizing.scale600",
  "paddingRight": "$theme.sizing.scale600",
  "paddingTop": "$theme.sizing.scale600",
}
`;

exports[`Component styled components StyledItem - dragged: StyledItem has correct styles when being dragged 1`] = `
Object {
  ":hover": Object {},
  "alignItems": "center",
  "backgroundColor": "$theme.colors.mono100",
  "borderBottomColor": "$theme.colors.primary",
  "borderBottomStyle": "solid",
  "borderBottomWidth": "2px",
  "borderLeftColor": "$theme.colors.primary",
  "borderLeftStyle": "solid",
  "borderLeftWidth": "2px",
  "borderRightColor": "$theme.colors.primary",
  "borderRightStyle": "solid",
  "borderRightWidth": "2px",
  "borderTopColor": "$theme.colors.primary",
  "borderTopStyle": "solid",
  "borderTopWidth": "2px",
  "boxShadow": "0px 2px 6px rgba(39, 110, 241, 0.32)",
  "cursor": "grabbing",
  "display": "flex",
  "justifyContent": "space-between",
  "listStyle": "none",
  "paddingBottom": "$theme.sizing.scale600",
  "paddingLeft": "$theme.sizing.scale600",
  "paddingRight": "$theme.sizing.scale600",
  "paddingTop": "$theme.sizing.scale600",
}
`;

exports[`Component styled components StyledItem - selected: StyledItem has correct styles when being selected 1`] = `
Object {
  ":hover": Object {},
  "alignItems": "center",
  "backgroundColor": "$theme.colors.mono100",
  "borderBottomColor": "$theme.colors.primary",
  "borderBottomStyle": "solid",
  "borderBottomWidth": "2px",
  "borderLeftColor": "$theme.colors.primary",
  "borderLeftStyle": "solid",
  "borderLeftWidth": "2px",
  "borderRightColor": "$theme.colors.primary",
  "borderRightStyle": "solid",
  "borderRightWidth": "2px",
  "borderTopColor": "$theme.colors.primary",
  "borderTopStyle": "solid",
  "borderTopWidth": "2px",
  "boxShadow": null,
  "cursor": "grab",
  "display": "flex",
  "justifyContent": "space-between",
  "listStyle": "none",
  "paddingBottom": "$theme.sizing.scale600",
  "paddingLeft": "$theme.sizing.scale600",
  "paddingRight": "$theme.sizing.scale600",
  "paddingTop": "$theme.sizing.scale600",
}
`;

exports[`Component styled components StyledLabel - basic render: StyledLabel has correct styles 1`] = `
Object {
  "flexGrow": 1,
  "fontFamily": "$theme.typography.font400.fontFamily",
  "fontSize": "$theme.typography.font400.fontSize",
  "fontWeight": "$theme.typography.font400.fontWeight",
  "lineHeight": "$theme.typography.font400.lineHeight",
}
`;

exports[`Component styled components StyledList - basic render: StyledList has correct styles 1`] = `
Object {
  "cursor": null,
  "paddingLeft": 0,
}
`;

exports[`Component styled components StyledList - dragging is active: StyledList has correct styles when dragging is active 1`] = `
Object {
  "cursor": "grabbing",
  "paddingLeft": 0,
}
`;

exports[`Component styled components StyledRoot - basic render: StyledRoot has correct styles 1`] = `
Object {
  "color": "$theme.colors.mono1000",
  "width": "100%",
}
`;
